dev/src/Command/AddComponentCommand.php

```diff
@@ -184,21 +213,160 @@ protected function execute(InputInterface $input, OutputInterface $output)
+        }
+
         $output->writeln('');
         $output->writeln('');
         $output->writeln('Success!');
 
         return 0;
     }
 
-    private function loadProtoContent(string $proto): string
+    private function loadRawFileContent(string $file): string
     {
-        if (file_exists($proto)) {
-            return file_get_contents($proto);
+        if (file_exists($file)) {
+            return file_get_contents($file);
+        }
+        if (count(explode('googleapis/', $file)) == 1) {
+            $fileUrlPath = $file;
+        } else {
+            $fileUrlPath = explode('googleapis/', $file)[1];
         }
-        $protoUrl = 'https://raw.githubusercontent.com/googleapis/googleapis/master/' . $proto;
+        $fileUrl = 'https://raw.githubusercontent.com/googleapis/googleapis/master/' . $fileUrlPath;
         $client = new Client();
-        $response = $client->get($protoUrl);
+        try {
+            $response = $client->get($fileUrl);
+        } catch (\Exception $e) {
+            // Handle error gracefully
+            return '';
+        }
         return (string) $response->getBody();
     }
+
+    private function bazelBuildLibrary(string $libraryName, string $googleApisDir, string $protoPath): string
+    {
+        $command = ['bazel', '--version'];
+        $output = $this->runCommand($command);
+        $component = explode($libraryName, $protoPath)[0];
+        // Extract the version number from the output
+        $match = preg_match('/bazel\s+(?P<version>\d+\.\d+\.\d+)/', $output, $matches);
+        if (!$match || $matches['version'] !== self::BAZEL_VERSION) {
+            throw new RuntimeException('Bazel 6.0.0 is not available');
+        }
+        $command = [
+            'bazel',
+            'query',
+            'filter("-(php)$", kind("rule", //' . $component . $libraryName .'/...:*))'
+        ];
+        $output = $this->runCommand($command, $googleApisDir);
+        $command = ['grep', '-v', '-E', ':(proto|grpc|gapic)-.*-php$'];
+        $output = $this->runCommand($command, $googleApisDir, $output);
+        // @TODO: Some non printable characters are included in the output, why!
+        $output = preg_replace('/[^[:print:]]/', '', $output);
+        $command = ['bazel', 'build', $output];
+        return $this->runCommand($command, $googleApisDir, null, 'errorOutput');
+    }
+
+    private function copyBazelOutToGoogleCloudPhp(string $componentName, string $googleApisDir): string
+    {
+        $command = ['which', 'docker'];
+        $output = $this->runCommand($command);
+        if (strlen($output) == 0) {
+            throw new RuntimeException(
+                'Error: Docker is not available.'
+            );
+        }
+        list($userId, $groupId) = $this->getUserAndGroupId();
+        $command = [
+            'docker',
+            'run',
+            '--rm',
+            '--user',
+            sprintf('%s::%s', $userId, $groupId),
```

Comment:
is it necessary to pass the user and group?

Reply:
Hmm.. I used to pass when I do the docker run, without this also everything is working fine, may be I will remove this.

Reply:
Removed

Reply:
I see a problem with this

Reply:
Actually, without --user, the permissions are messed up: `drwxr-xr-x  3 root     root`. Will add this back.